Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -1118,14 +1137,21 @@ static bool isCopyDisplayedPixelBufferAvailable()
             return;
         weakThis->didProvideMediaDataForTrackId(WTFMove(sample), trackId, mediaType);
     });
-    m_parser->setCallOnClientThreadCallback([](auto&& function) {
-        function();
-    });
 
     m_parsingSucceeded = true;
+    m_pendingAppends++;
 
     SourceBufferParser::Segment segment(Ref { buffer });
-    m_parser->appendData(WTFMove(segment));
+    m_appendQueue->dispatch([segment = WTFMove(segment), weakThis = WeakPtr { *this }, parser = m_parser, abortCalled = m_abortCalled]() mutable {
+        parser->appendData(WTFMove(segment), [weakThis = WTFMove(weakThis), abortCalled]() mutable {
```

Comment:
Shouldn't this check `weakThis`, won't `parser` be null if it is?

Reply:
We wait for all dispatched functions to be executed in our destructor, so I don't think it's possible for the dispatched function to run while MPPWebM is destroyed. Thoughts?

Reply:
I guess we can check for `abortCalled` so we don't unnecessarily append.

Reply:
Even if you think it can't be NULL it is probably worth adding an ASSERT